### DISCUSSION TITLE
Add trigger/sensor to samba provider and make it deferrable

### DISCUSSION THREAD

**Discussion by jertel-etr** [apache/airflow#36334] [Ideas]
I have been working with the samba provider a lot, as I work for an organization that exports a lot of files to on prem network storage that needs ingested into the cloud. I would be interested in looking at using the aiosmb library  in order to make it so that the hook can be used in deferrable sensors/triggers, and then adding in a file sensor and trigger into the provider. This is something I would be interested in working on, but would like to solicit feedback from the community.

**Comment by Taragolis** [comment]
> but would like to solicit feedback from the community.

Feel free make a PR. You don't have to wait for someone to tell you "Awesome idea. lets implement it", it would be enough that personally you think that this might be awesome idea.


Some useful links which might help you to make first contribution into the Apache Airflow:
- [Contributor's Quick Start](https://github.com/apache/airflow/blob/main/CONTRIBUTORS_QUICK_START.rst#contributors-quick-start)
- [CONTRIBUTING.rst](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst)
- [BREEZE.rst](https://github.com/apache/airflow/blob/main/BREEZE.rst)

In addition you could ask for for help with the first contribution in Apache Airflow Community Slack ([join link](https://s.apache.org/airflow-slack), [workspace link](https://apache-airflow.slack.com/)) channel `#development-first-pr-support`
